config/initializers/subjects/course_subjects.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require "degree_subjects"
+
+unless defined?(CourseSubjects)
```

Comment:
Not ideal, but it allowed me to get around issues rails 7 had loading these constants in other directories such as `lib/` and not doing it in a way that raises warnings about constants are loaded.